salt/orch/removal.sls

```diff
@@ -0,0 +1,151 @@
+       - update-grains
+
+update-modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+    - require:
+      - update-mine
+
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master', fun='network.interfaces', tgt_type='compound').keys() %}
+{%- if target in masters %}
+  # TODO: we should promote a worker, but this is too hard to do right now
+  #       just log a message for now
+  {%- do salt.caasp_log.warn('removing a k8s master: it will not be replaced') %}
+{%- endif %}
+
+{%- set etcd_members = salt.saltutil.runner('mine.get', tgt='G@roles:etcd', fun='network.interfaces', tgt_type='compound').keys() %}
+{%- if target in etcd_members %}
+  # if the node is a member of the etcd cluster:
+  # we must choose another node and promote it to be an etcd master
+  {%- set non_etcd_members = salt.saltutil.runner('mine.get', tgt='not G@roles:etcd', fun='network.interfaces', tgt_type='compound').keys() %}
+  {%- if non_etcd_members|length > 0 %}
+    {%- set new_etcd_member = non_etcd_members|first %}
+    # promote some other node to be a full member of the etcd cluster
+    # WARNING: we should do a consistency check in Velum...
+etcd-add-member:
+  salt.state:
+    - tgt: '{{ new_etcd_member }}'
+    - sls:
+      - etcd.remove-pre-stop-services
+    - require:
+      - update-modules
+  {%- endif %}
+
+  # remove the etcd member running in {{ target }}
+etcd-remove-member:
+  salt.state:
+    - tgt: '{{ target }}'
+    - sls:
+      - etcd.remove-pre-stop-services
+    - require:
+    {%- if non_etcd_members|length > 0 %}
+      - etcd-add-member
+    {%- else %}
+      - update-modules
+    {%- endif %}
+
+{%- endif %}
+
+update-grains-again:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+    - require:
+    {%- if target in etcd_members %}
+      - etcd-remove-member
+    {%- else %}
+      - update-modules
+    {%- endif %}
+
+stop-services:
+  salt.state:
+    - tgt: {{ target }}
+    - sls:
+      - container-feeder.stop
+      {%- if target in masters %}
+      - kube-apiserver.stop
+      - kube-controller-manager.stop
+      - kube-scheduler.stop
+      {%- else %}
+      - kubelet.stop
+      - kube-proxy.stop
+      {%- endif %}
+      - docker.stop
+      - etcd.stop
+    - require:
+      - update-grains-again
+
+# remove any other configuration in the machines
+# TODO: all the files in /etc/kubernetes/*, /var/lib/kubernetes/* ??
+remove-pre-reboot:
+  salt.state:
+    - tgt: {{ target }}
+    - sls:
+      - container-feeder.remove-pre-reboot
+      {%- if target in masters %}
+      - kube-apiserver.remove-pre-reboot
+      - kube-controller-manager.remove-pre-reboot
+      - kube-scheduler.remove-pre-reboot
+      {%- else %}
+      - kubelet.remove-pre-reboot
+      - kube-proxy.remove-pre-reboot
+      {%- endif %}
+      - docker.remove-pre-reboot
+      - cni.remove-pre-reboot
+      - etcd.remove-pre-reboot
```

Comment:
Please add also 
```
- etc-hosts.remove-pre-reboot
```
I've added it to https://github.com/kubic-project/salt/pull/369